### PR TITLE
Simplify digitization request list

### PR DESCRIPTION
**Pull request by eddierubeiz**
Ref #2561.
User interface works for @Gabz73 and @JAuerbach26.
I added extra controller tests, too.

### CODE REVIEW

**Comment by jrochkind** [comment]
oh wow this is a lot of files changed. I'm going to actually check it out to play with it to review

**Comment by jrochkind** [comment]
When we last talked to @Gabz73, my understanding was they still wanted the current view by collecting area but ALSO wanted a view that listed everything regardless of collecting area.  As recorded in [this comment](https://github.com/sciencehistory/scihist_digicoll/issues/2561#issuecomment-1984568882)

So I had imagined we'd still have the home screen by a list of collecting areas that you could click through, to support that previous workflow where staff was working focused on a specific collecting area. 

But in this PR, that has been removed, and you only get one "main" list of queue items. You can choose from a popup menu to restrict by collecting area, a little bit different UI. 

@Gabz73 can you confirm this UI is what you wanted, you wanted to get rid of that list by collecting area as initial screen, it's no longer best for current workflow?

When we **are** looking at the list by all collecting areas -- the initial list screen does _not_ have "collecting area" on it, you don't know what collecting area an item is in unless you click through. Is this okay?

This is more change to UI than I had anticipated, just want to make sure these changes were intentional responses to new workflow needs?  Do we want to like user test it with digitization staff or anything?

Also, I note that there's still a "Rights & Reproduction" checkbox in digitization item metadata -- I no longer remmeber what this does or how these even show up -- this is still used, and has been tested to ensure it still works as desired with these changes? @eddierubeiz @Gabz73

File: app/models/admin/digitization_queue_item.rb [hunk 1]
```diff
@@ -23,6 +23,7 @@ def validate_each(record, attribute, value)
   COLLECTING_AREAS = %w{archives rare_books modern_library museum}
 
   validates :collecting_area, inclusion: { in: COLLECTING_AREAS }
+  validates :collecting_area, presence: true
```
Comment (jrochkind) [inline app/models/admin/digitization_queue_item.rb:26]: why this change, no longer making sure it is in the allow list of allowed collections?

**Comment by Gabz73** [comment]
I may not have expressed it well in the original request, but this is what we want:  you can see the entire queue, by order of deadline, or you can filter the queue to a particular collecting area. When you are in the "entire queue" view, you can tell what collection something belongs to by virtue of the parenthetical qualifier. So, yes, can confirm that we don't need the initial screen that shows the list by collecting area. 



>Do we want to like user test it with digitization staff or anything?

**This is a good reminder that Sarah Newhouse is also adding items to the DC. I spoke with her, and showed her the draft queue page, which she had no issues with. Jahna has already seen it.  That's all the dig staff!**



>Also, I note that there's still a "Rights & Reproduction" checkbox in digitization item metadata -- I no longer remember what this does or how these even show up -- this is still used, and has been tested to ensure it still works as desired with these changes? @eddierubeiz @Gabz73

It is still used, and yes it works.

**Comment by jrochkind** [comment]
Thanks @Gabz73 !

File: app/models/admin/digitization_queue_item.rb (hunk 1 above)
Comment (eddierubeiz) [reply app/models/admin/digitization_queue_item.rb]: This was a mistake - thanks for catching it!
